lib/lightning_web/live/project_live/settings.ex

```diff
@@ -43,6 +44,42 @@ defmodule LightningWeb.ProjectLive.Settings do
         socket.assigns.project
       )
 
+    repo_connection =
+      VersionControl.get_repo_connection(socket.assigns.project.id)
+
+    # hide github install if our project has a repo connection
+    show_github_setup =
+      case repo_connection do
+        nil -> true
+        _ -> false
+      end
+
+    show_repo_setup =
+      case repo_connection do
+        nil ->
+          false
+
+        conn ->
+          case conn.repo do
+            nil -> true
+            _ -> false
+          end
+      end
+
+    show_sync_button =
+      case repo_connection do
+        nil ->
+          false
+
+        conn ->
+          case conn.repo && conn.branch do
+            nil -> false
+            _ -> true
+          end
+      end
+
```

Comment:
Could these not be better expressed using either, `is_nil`, `repo_connection && repo_connection.repo` etc? Either that or a `cond`?